File: algorithm/001-two_sum/python3/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        complements = {}
        for i in range(len(nums)):
            c = target - nums[i]
            if c in complements:
                return [complements[c], i]
            complements[nums[i]] = i
        return []

File: algorithm/001-two_sum/go/solution.go
package solution

func twoSum(nums []int, target int) []int {
	var complements = map[int]int{}
	for i := 0; i < len(nums); i++ {
		var c = target - nums[i]
		if complements[c] > 0 {
			return []int{complements[c] - 1, i}
		}
		complements[nums[i]] = i + 1
	}

	return []int{}
}
